docs/source/reference/ai/custom-ai-function.rst

```diff
@@ -39,47 +41,102 @@ An abstract method that must be implemented in your child class. The setup funct
 
-- input_signatures: List[IOColumnArgument] 
+- input_signatures: List[IOArgument]
    
   Data types of the inputs to the forward function must be specified. If no constraints are given, then no validation is done for the inputs.
 
-- output_signatures: List[IOColumnArgument]
+- output_signatures: List[IOArgument]
 
   Data types of the outputs to the forward function must be specified. If no constraints are given, then no validation is done for the inputs.
 
-A sample forward function is given below
+A list of IO types in EvaDB are found in `IODescriptors <https://github.com/georgia-tech-db/evadb/blob/staging/evadb/functions/decorators/io_descriptors/data_types.py>`_ folder.
+
+Please ensure that the names of the columns in the dataframe match the names specified in the decorators.
+
+
+Part 2: Registering and using the function in EvaDB Queries
+-----------------------------------------------------------
+
+Now that you have implemented your function, we need to register it as a function in EvaDB. You can then use the function in any query.
+
+1. Register the function with a query that follows this template:
+
+    `CREATE FUNCTION [ IF NOT EXISTS ] <name>
+    IMPL <path_to_implementation>;`
+
+  where,
+
+        * <name> - specifies the unique identifier for the function.
+        * <path_to_implementation> - specifies the path to the implementation class for the function
+
+
+2. Execute your function on any data:
+
+  .. code-block:: sql
+
+      SELECT <name>(data) FROM table [condition];
+
+3. You can drop the function when you no longer need it.
+
+  .. code-block:: sql
+
+      DROP FUNCTION IF EXISTS <name>;
+----------
+
+Examples
+---------
+
+1. Yolo Object Detection
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following code can be used to create an Object Detection function using Yolo
+
+1. Inherit from the `Abstract Function` class. It is a parent class that defines and implements standard methods for model inference.
+
+    .. code-block:: python
+  
+        class Yolo(AbstractFunction)
 
-.. code-block:: python
+2. Create the setup function. Here we set the parameters batchable as True, cacheable as True and function type as 'object_detection'. The function has code to perform some basic initialization for the Yolo function. 
+
+    .. code-block:: python
+
+      @setup(cacheable=True, function_type="object_detection", batchable=True)
+      def setup(self, threshold=0.85):
+         try_to_import_ultralytics() #function to try and import the YOLO library.
+         from ultralytics import YOLO
     
-    @forward(
-          input_signatures=[
-              PyTorchTensor(
-                  name="input_col",
-                  is_nullable=False,
-                  type=NdArrayType.FLOAT32,
-                  dimensions=(1, 3, 540, 960),
+          self.threshold = threshold #sets the threshold for the model
+          self.model = YOLO(model) #initializes the model
+          self.device = "cpu" #sets the device as CPU
```

Comment:
Is GPU case for interesting for the users?

Reply:
I am not sure about this. Since most people run it on the local they mostly use CPU.